app/assets/javascripts/admin/views/dataset/dataset_upload_view.js

```diff
@@ -133,15 +128,14 @@ class DatasetUploadView extends Marionette.View {
         }),
       )
       .then(
-        ({ messages }) => {
-          Toast.message(messages);
-          app.router.navigate("/dashboard", { trigger: true });
+        () => {
+          Toast.success(messages["dataset.upload_success"]);
+          const url = `/datasets/${form.name.value}/import`;
+          app.router.navigate(url, { trigger: true });
         },
-        () => {}, // NOOP
-      )
-      .then(
-        () =>  // always do
-           this.ui.spinner.addClass("hidden"),
+        () => {
+           this.ui.spinner.addClass("hidden");
```

Comment:
Yes. The `Request` library defaults to show an error toast on unsuccessful AJAX calls.